openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -239,70 +239,67 @@ def is_flag_active(self, flag_name, check_before_waffle_callback=None, flag_unde
-                # NOTE: This will be removed once ARCHBOM-132, currently in-progress, is complete.
-                'flag_undefined_default has been deprecated. For existing uses this is already actively being fixed.',
-                DeprecationWarning
-            )
 
+        """
         # validate arguments
         namespaced_flag_name = self._namespaced_name(flag_name)
-        value = None
+
         if check_before_waffle_callback:
             value = check_before_waffle_callback(namespaced_flag_name)
-
-        if value is None:
-            # Do not get cached value for the callback, because the key might be different.
-            # The callback needs to handle its own caching if it wants it.
-            value = self._cached_flags.get(namespaced_flag_name)
-            if value is None:
-                is_flag_active_for_everyone = False
-                if flag_undefined_default is not None:
-                    # determine if the flag is undefined in waffle
-                    try:
-                        waffle_flag = Flag.objects.get(name=namespaced_flag_name)
-                        is_flag_active_for_everyone = (waffle_flag.everyone is True)
-                    except Flag.DoesNotExist:
-                        if flag_undefined_default:
-                            # This metric will go away once this has been fully retired with ARCHBOM-132.
-                            # Also, even though the value will only track the last flag, that should be enough.
-                            set_custom_metric('temp_flag_default_used', namespaced_flag_name)
-                        value = flag_undefined_default
-
-                if value is None:
-                    request = crum.get_current_request()
-                    if request:
-                        value = flag_is_active(request, namespaced_flag_name)
-                    else:
-                        log.warning(u"%sFlag '%s' accessed without a request", self.log_prefix, namespaced_flag_name)
-                        # Return the Flag's Everyone value if not in a request context.
-                        # Note: this skips the cache as the value might be different
-                        # in a normal request context. This case seems to occur when
-                        # a page redirects to a 404, or in Celery workers.
-                        self._set_waffle_flag_metric(namespaced_flag_name, is_flag_active_for_everyone)
-                        set_custom_metric('warn_flag_no_request_return_value', is_flag_active_for_everyone)
-                        return is_flag_active_for_everyone
-
-                self._cached_flags[namespaced_flag_name] = value
+            if value is not None:
+                # Do not cache value for the callback, because the key might be different.
+                # The callback needs to handle its own caching if it wants it.
+                self._set_waffle_flag_metric(namespaced_flag_name, value)
+                return value
```

Comment:
**Reviewer Note:** I refactored to return earlier and hopefully the newer code is simpler to read.  
Please pay close attention to help ensure I didn't break anything, although the unit tests seem to be happy. :)